src/porepy/models/constitutive_laws.py

```diff
@@ -676,35 +676,58 @@ class CubicLawPermeability(ConstantPermeability):
 
     """
 
-    def permeability(self, subdomains: list[pp.Grid]) -> np.ndarray:
+    def permeability(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
         """Permeability [m^2].
 
         Parameters:
-            subdomain: Subdomain where the permeability is defined. Should be of length
-                1; the list format is used for compatibility with similar methods.
+            subdomains: List of subdomains.
 
         Returns:
-            Cell-wise permeability tensor. The value is picked from the solid constants
-                for the matrix and computed from the fracture aperture for fractures and
-                intersections.
-
-        Raises:
-            ValueError: If more than one subdomain is provided.
+            Cell-wise permeability values.
 
         """
-        if len(subdomains) > 1:
-            raise ValueError("Only one subdomain is allowed.")
+        projection = pp.ad.SubdomainProjections(subdomains, dim=1)
+        matrix = [sd for sd in subdomains if sd.dim == self.nd]
+        fractures_and_intersections: list[pp.Grid] = [
+            sd for sd in subdomains if sd.dim < self.nd
+        ]
+
+        permeability = projection.cell_prolongation(matrix) * self.matrix_permeability(
+            matrix
+        ) + projection.cell_prolongation(
+            fractures_and_intersections
+        ) * self.cubic_law_permeability(
+            fractures_and_intersections
+        )
+        return permeability
+
+    def cubic_law_permeability(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Cubic law permeability for fractures (or intersections).
 
-        if subdomains[0].dim == self.nd:
-            return super().permeability(subdomains)
+        Parameters:
+            subdomains: List of subdomains.
+
+        Returns:
+            Cell-wise permeability operator.
 
-        # Fracture or intersection
+        """
         aperture = self.aperture(subdomains)
-        perm = aperture**2 / 12
-        # Scale by specific volume
-        perm = (perm * self.specific_volume(subdomains)).evaluate(self.equation_system)
+        perm = aperture**2 * pp.ad.Scalar(1 / 12)
+
         return perm
 
+    def matrix_permeability(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Permeability of the matrix.
+
+        Parameters:
+            subdomains: List of subdomains.
+
+        Returns:
+            Cell-wise permeability operator.
+
+        """
+        return super().permeability(subdomains)
```

Comment:
I can see this `super` leading to trouble for someone who makes a non-trivial combination of permeability classes, but I believe that risk is unavoidable at some level. I'm fine with keeping this.

Reply:
I think you're right. But the trouble should be avoidable by overriding `matrix_permeability` in that combination.